src/cascadia/Remoting/WindowManager.cpp

```diff
@@ -103,24 +107,58 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
                                   "WindowManager_ProposeCommandline",
                                   TraceLoggingBoolean(_shouldCreateWindow, "CreateWindow", "true iff we should create a new window"),
                                   TraceLoggingPointer(nullptr, "Id", "No ID provided"),
+                                  TraceLoggingWideString(givenName.c_str(), "Name", "The name we should assign this window"),
                                   TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
             }
         }
         else
         {
             // We're the monarch, we don't need to propose anything. We're just
             // going to do it.
-            TraceLoggingWrite(g_hRemotingProvider,
-                              "WindowManager_ProposeCommandline_AsMonarch",
-                              TraceLoggingBoolean(_shouldCreateWindow, "CreateWindow", "true iff we should create a new window"),
-                              TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+            //
+            // However, we _do_ need to ask what our name should be. It's
+            // possible someone started the _first_ wt with something like `wt
+            // -w king` as the commandline - we want to make sure we set our
+            // name to "king".
+            auto findWindowArgs{ winrt::make_self<Remoting::implementation::FindTargetWindowArgs>(args) };
+            _raiseFindTargetWindowRequested(nullptr, *findWindowArgs);
+
+            const auto responseId = findWindowArgs->ResultTargetWindow();
+            if (responseId > 0)
+            {
+                givenID = ::base::saturated_cast<uint64_t>(responseId);
+
+                TraceLoggingWrite(g_hRemotingProvider,
+                                  "WindowManager_ProposeCommandline_AsMonarch",
+                                  TraceLoggingBoolean(_shouldCreateWindow, "CreateWindow", "true iff we should create a new window"),
+                                  TraceLoggingUInt64(givenID.value(), "Id", "The ID we should assign our peasant"),
+                                  TraceLoggingWideString(givenName.c_str(), "Name", "The name we should assign this window"),
+                                  TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE));
+            }
+            else if (responseId == WindowingBehaviorUseName)
+            {
+                givenName = findWindowArgs->ResultTargetWindowName();
+
+                TraceLoggingWrite(g_hRemotingProvider,
+                                  "WindowManager_ProposeCommandline_AsMonarch",

```

Comment:
prefer not having two events with the same name but different payloads. Hard to make a database table based on it. You can just have one event with every state in it.